docs/user-guide/plugins.md

```diff
@@ -229,6 +229,15 @@ entire site.
     Parameters:
     : __config:__ global configuration object
 
+##### on_build_error
+
+:   The `build_error` event is called after an error happened during the build.
+    It is triggered by plugins by raising a `mkdocs.exceptions.PluginError`.
+    Use this event to clean things up before the build is aborted.
```

Comment:
This is not exactly accurate. Saying that the "event is called after an error happened" implies that it is called immediately after. Instead it is called at the end of the build process. I realize that technically, the error short circuits the build process, but I think this could be a little clearer. 

For example: "the event is only called at the end of the build process when a `PluginError` has been raised during the build." I realize that's a little rough, but I think it is a better general direction.

Reply:
But *it is* called immediately after :smile: 

Indeed, raising a `PluginError` will interrupt every scopes of execution until it is caught. We wrapped the whole `build` function's body into a `try`/`except PluginError`, and we immediately trigger the `build_error` event when such an exception is caught. The build process actually never ends as it is interrupted by the exception. Now it seems that you know that, based on your second comment about short-circuiting the build process, so I'm not sure what you first meant :confused: 

Maybe you meant that it is *physically* called at the end of the build process, as in, based on line numbers? But do plugin developers want/need to know that?

Lastly, your example would make me think that the error is somehow recorded, and that we check at the end of the build process if there was an error. But this is not what happens, as we do not record it, but rather catch it and immediately process it. It would also make me think that I, as a plugin developer, could raise multiple `PluginError`, letting the build continue, and have it exit with a code greater than 0. This would be incorrect again (except for the exit code).

I'm just not sure to understand what you mean :slightly_smiling_face:

Reply:
> Maybe you meant that it is physically called at the end of the build process,

Yes, that is what I mean. The way you worded it, that was not clear. I'm asking that that be made clear.

Reply:
OK, I'll try to rephrase then :slightly_smiling_face: